### DISCUSSION TITLE
Adding sanity checks for functions that requires TLS or DTLS to work

### DISCUSSION THREAD

**Discussion by fwh-dc** [openssl/openssl#23132] [Q&A]
I see some uses of f.ex. `SSL3_HM_HEADER_LENGTH` with no checks that the function is only called for TLS connections. I was considering adding some additional assertion checks for those functions. What's your opinion?

And also: which assert function do you prefer? I see `assert()` and `OPENSSL_assert()` are both in used throughout the codebase.

**Comment by mattcaswell** [comment]
The coding style guide was recently updated with guidance on this. See chapter 16:

https://www.openssl.org/policies/technical/coding-style.html

We should be avoiding `OPENSSL_assert()`. It should only be used where *global* state is corrupted.

In general prefer `ossl_assert`.

**Reply by fwh-dc** [reply]
Nice, thanks for info.
Do you think it is worthwhile adding the sanity checks to the codebase as I suggest?

**Reply by mattcaswell** [reply]
Seems like it could be a good idea.
